src/core/coap/coap_header.cpp

```diff
@@ -281,6 +312,56 @@ ThreadError Header::AppendContentFormatOption(MediaType aType)
     return AppendOption(coapOption);
 }
 
+ThreadError Header::AppendMaxAgeOption(uint32_t aMaxAge)
+{
+    Option coapOption;
+    uint8_t buf[4];
+
+    coapOption.mNumber = kCoapOptionMaxAge;
+
+    if (aMaxAge <= 0xff)
```

Comment:
Have you considered using a loop to make the code more concise? You could make it like:
```
aMaxAge = Encoding::BigEndian::HostSwap32(aMaxAge);
coapOption.mLength = 0;
do
{
    buf[coapOption.mLength] = aMaxAge & 0xff;
    aMaxAge = aMaxAge >> 8;
    coapOption.mLength++;
} while (aMaxAge != 0 && coapOption.mLength < sizeof(buf))
```
You could even make an generic function from it and use in both, AppendMaxAgeOption and AppendObserveOption.